### PR TITLE
Feature/poam link

### PR DESCRIPTION
**Pull request by wandmagic**
# Committer Notes

adding some poam item related constraints

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [x] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Comment by wandmagic** [comment]
see #934

File: src/validations/constraints/content/ssp-component-has-poam-link-INVALID.xml [hunk 1]
```diff
@@ -0,0 +1,1591 @@
+<system-security-plan xmlns="http://csrc.nist.gov/ns/oscal/1.0" uuid="11111111-2222-4000-8000-000000000000">
```
Comment (Gabeblis) [inline src/validations/constraints/content/ssp-component-has-poam-link-INVALID.xml:1]: Are we able to trim down this file to the minimum necessary content?

File: src/validations/constraints/content/ssp-poam-link-has-resource-fragment-INVALID.xml [hunk 2]
```diff
@@ -0,0 +1,1592 @@
+<system-security-plan xmlns="http://csrc.nist.gov/ns/oscal/1.0" uuid="11111111-2222-4000-8000-000000000000">
```
Comment (Gabeblis) [inline src/validations/constraints/content/ssp-poam-link-has-resource-fragment-INVALID.xml:1]: Are we able to trim down this file to the minimum necessary content?

File: src/validations/constraints/content/ssp-poam-link-references-valid-resource-INVALID.xml [hunk 3]
```diff
@@ -0,0 +1,403 @@
+<system-security-plan xmlns="http://csrc.nist.gov/ns/oscal/1.0" uuid="11111111-2222-4000-8000-000000000000">
```
Comment (Gabeblis) [inline src/validations/constraints/content/ssp-poam-link-references-valid-resource-INVALID.xml:1]: Are we able to trim down this file to the minimum necessary content?

File: src/validations/constraints/content/ssp-poam-resource-has-oscal-link-INVALID.xml [hunk 4]
```diff
@@ -0,0 +1,1592 @@
+<system-security-plan xmlns="http://csrc.nist.gov/ns/oscal/1.0" uuid="11111111-2222-4000-8000-000000000000">
```
Comment (Gabeblis) [inline src/validations/constraints/content/ssp-poam-resource-has-oscal-link-INVALID.xml:1]: Are we able to trim down this file to the minimum necessary content?

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 5]
```diff
@@ -740,4 +746,54 @@
         </constraints>
     </context>
 
+    <context>
```
Comment (Gabeblis) [inline src/validations/constraints/fedramp-external-constraints.xml:749]: This doesn't comply with [frr102](https://github.com/GSA/fedramp-automation/blob/develop/src/validations/styleguides/STYLE.md/#FRR102). Can we move this into the existing context block with the `/system-security-plan` target.

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 6]
```diff
@@ -740,4 +746,54 @@
...
+                <description>Index of all POAM items in the referenced POAM document</description>
+                <key-field target="@uuid"/>
+            </index>
+            <expect id="ssp-poam-resource-has-oscal-link" target="." test="exists($poam-uri)" level="ERROR">
+                <formal-name>SSP Has POAM Link</formal-name>
+                <description>Each SSP MUST have a link to a valid OSCAL POAM</description>
+                <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="https://automate.fedramp.gov/documentation/poam/4-poam-template-to-oscal-mapping/"/>
+                <message>Each FEDRAMP SSP MUST have a link to a valid OSCAL POAM.</message>
+            </expect>
+            <expect id="ssp-component-has-poam-link" target="$network-components" test="exists($poam-uri) and count(link[@rel='poam-item']) = 1" level="ERROR">
```
Comment (Gabeblis) [inline src/validations/constraints/fedramp-external-constraints.xml:770]: These constraints don't comply with [frr103](https://github.com/GSA/fedramp-automation/blob/develop/src/validations/styleguides/STYLE.md#frr103).

**Review by Gabeblis — CHANGES_REQUESTED** [review]
Good stuff. Just a couple things related to the style guide and to help make this PR easier to look at.

**Comment by wandmagic** [comment]
@Gabeblis  can you take a look? i added some more checks so we don't have to check alphabetical order, or duplicate constraints

**Comment by wandmagic** [comment]
converting to draft while i pull out style guide edits

**Comment by aj-stein-gsa** [comment]
Are you reopening this? I also get part-way through a PR and then have to start over with reviews. 😅

**Pull request closed without merging**